test/gtest/uct/test_md.cc

```diff
@@ -917,6 +917,58 @@ UCS_TEST_SKIP_COND_P(test_md, exported_mkey,
     ASSERT_UCS_OK(status);
 }
 
+UCS_TEST_P(test_md, rkey_compare_params_check)
+{
+    uct_rkey_compare_params_t params = {};
+    ucs_status_t status;
+    int result;
+
+    status = uct_rkey_compare(GetParam().component, 0, 0, &params, NULL);
+    ASSERT_UCS_STATUS_EQ(UCS_ERR_INVALID_PARAM, status);
+
+    params.field_mask = 1;
+    status = uct_rkey_compare(GetParam().component, 0, 0, &params, &result);
+    ASSERT_UCS_STATUS_EQ(UCS_ERR_INVALID_PARAM, status);
+}
+
+// SM case is covered by XPMEM which has registration capability
+UCS_TEST_SKIP_COND_P(test_md, rkey_compare, !check_caps(UCT_MD_FLAG_REG))
+{
+    size_t size                      = 4096;
+    void *address                    = NULL;
+    uct_rkey_compare_params_t params = {};
+    std::vector<uint8_t> rkey_buffer(md_attr().rkey_packed_size + 1);
+    uct_rkey_bundle_t b1, b2;
+    ucs_status_t status;
+    uct_mem_h memh;
+    int result, expect_result;
+
+    status = ucs_mmap_alloc(&size, &address, 0, "test_rkey_compare_equal");
+    ASSERT_UCS_OK(status);
+    status = reg_mem(UCT_MD_MEM_ACCESS_ALL, address, size, &memh);
+    ASSERT_UCS_OK(status);
+    status = uct_md_mkey_pack(md(), memh, &rkey_buffer[0]);
+    ASSERT_UCS_OK(status);
+
+    status = uct_rkey_unpack(GetParam().component, &rkey_buffer[0], &b1);
+    ASSERT_UCS_OK(status);
+    status = uct_rkey_unpack(GetParam().component, &rkey_buffer[0], &b2);
+    ASSERT_UCS_OK(status);
+
+    status = uct_rkey_compare(GetParam().component, b1.rkey, b2.rkey, &params,
+                              &result);
+    EXPECT_UCS_OK(status);
+    expect_result = b1.rkey > b2.rkey ? 1 : b1.rkey < b2.rkey ? -1 : 0;
```

Comment:
this is too strict expectation.
1. expect that result of rkey1,rkey2 is the opposite of the result of rkey2,rkey1
2. also, for same memh: expect the result to be 0